controllers/networking-calico/pkg/charts/utils.go

```diff
@@ -21,20 +21,67 @@ import (
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 )
 
+const (
+	hostLocal  = "host-local"
+	usePodCIDR = "usePodCidr"
+)
+
 // ComputeCalicoChartValues computes the values for the calico chart.
 func ComputeCalicoChartValues(network *extensionsv1alpha1.Network, config *calicov1alpha1.NetworkConfig) map[string]interface{} {
-	return map[string]interface{}{
-		"images": map[string]interface{}{
-			calico.CNIImageName:             imagevector.CalicoCNIImage(),
-			calico.TyphaImageName:           imagevector.CalicoTyphaImage(),
-			calico.KubeControllersImageName: imagevector.CalicoKubeControllersImage(),
-			calico.NodeImageName:            imagevector.CalicoNodeImage(),
-		},
-		"config": map[string]string{
+	var (
+		calicoChartValues = map[string]interface{}{
+			"images": map[string]interface{}{
+				calico.CNIImageName:             imagevector.CalicoCNIImage(),
+				calico.TyphaImageName:           imagevector.CalicoTyphaImage(),
+				calico.KubeControllersImageName: imagevector.CalicoKubeControllersImage(),
+				calico.NodeImageName:            imagevector.CalicoNodeImage(),
+			},
+			"global": map[string]string{
+				"podCIDR": network.Spec.PodCIDR,
+			},
+		}
+	)
+
+	var configValues = map[string]interface{}{}
+	if config.IPAM != nil {
```

Comment:
```suggestion
	if config.IPAM != nil && len(config.IPAM.Type) > 0 {
```

Get rid of one level of nesting